### ISSUE TITLE
[BUG] Motion Buttons Don't work 

### ISSUE THREAD

**Issue by seta-san** [PCSX2/pcsx2#4239]
my apologies to refraction.
**Describe the bug**
No matter how much you wave over the motion buttons they don't respond ore not enough to activate

**To Reproduce**
Get to the menus of several eyetoy games

Groove
Kaiketsu Zorori: Mezase! Itazura King - Limited Edition
Play 
Play 3
Play Hero - needs something long and shiney green to use as a sword
Rhythmic Star
Harry Potter and the Prisoner of Azkaban
Operation Spy
Play 2
Astro Zoo - works with some difficulty
Play Sports - works with great difficulty activating menu items

**Expected behavior**
the buttons build a meter by waving your hands over them until they are filled up activating the desired action

**GS Settings**
all standard. i do my testing in software.

**Emulation Settings**
you can mitigate the issue by using the interpreter 

**GS Window Screenshots**
not graphical in nature

**System Info (please complete the following information):**
 - PCSX2 Revision: 1.7.0-dev-1003
 - OS: Windows 10
 - CPU: i5-9600
 - GPU: GTX 960

**Comment by Florin9doi** [comment]
Possible improvements to be investigated:
1. Don't start sending the same frame again, and send NAK instead until a new frame is available, VideoDevice::GetImage should delete mpeg_buffer. This way, the next frame wont be delayed until the previous duplicate is completely transferred.
2. Modify the MPEG encoder to output partial data as soon as an the equivalent of an USB packet is encoded (896 bytes). This way, the upper part of the image can be transferred before the encoding is finished.
3. Verify if using a high speed camera has any impact - PlayStation Eye configured at 100fps or more.

**Comment by refractionpcsx2** [comment]
> Possible improvements to be investigated:
> 
> 3. Verify if using a high speed camera has any impact - PlayStation Eye configured at 100fps or more.

Are you sure on that? As far as I can find it's a 25fps camera, maybe 30 in NTSC regions

https://askubuntu.com/questions/471732/ps2-eyetoy-recording-quality

**Comment by Florin9doi** [comment]
Yes. I mean PlayStation Eye, the PS3 camera.

**Comment by Florin9doi** [comment]
I checked a few games using a PS3 Eye@100fps and they start to work better.

**Comment by seta-san** [comment]
what refraction did in https://github.com/PCSX2/pcsx2/pull/4225 was to forcefully make the mpg2 decoder in pcsx2 treat every frame as an I-Frame if decoder.coding_type = ipuRegs.ctrl.PCT == 0 which an invalid frame type. I've been wondering why the emulator is sending a nonvalid type and i think it comes down to the mpg encoder in the eyetoy module. I'm not 100% sure that the mpg encoder actually encodes in mpg2 or just mpg1. I'm no expert in mpg encoding but i don't particularly see anything in it that allows for encoding P or B frames as they would involve looking at previous frames for both and next frames for B frames and motion vectors

i added to ipu.cpp
`case 1://Macroblock Type
					decoder.frame_pred_frame_dct = 1;
					**if (ipuRegs.ctrl.PCT == 0)
						IPU_LOG("MACROBLOCK SET TO ZERO");
					if (ipuRegs.ctrl.PCT == 1)
						IPU_LOG("MACROBLOCK SET TO ONE");
					if (ipuRegs.ctrl.PCT == 2)
						IPU_LOG("MACROBLOCK SET TO TWO");
					if (ipuRegs.ctrl.PCT == 3)
						IPU_LOG("MACROBLOCK SET TO THREE");**
					decoder.coding_type = ipuRegs.ctrl.PCT > 0 ? ipuRegs.ctrl.PCT : 1; // Kaiketsu Zorro Mezase doesn't set a Picture type, seems happy with I
					ipuRegs.cmd.DATA = get_macroblock_modes();
					break;`
every frame has come up with macroblock being 0. with refractions hack every frame is being treated like a I-Frame(uncompressed video with no motion data). i'm thinking a more complex mpg2 encoder is needed.

**Comment by Florin9doi** [comment]
>right now the frames are "IIIIIIIIIIIIIIIIIIIIIIIIIIIIII"

That's correct and that's how the original Eyetoy works. Its focus was to provide low latency video, not the best compression/size. 

>allows for encoding P or B frames as they would involve looking at previous frames for both and **next frames**

We're talking about a live webcam after all. It can't predict the future frames without adding a 500 ms delay using a memory buffer it doesn't have.

**Comment by Florin9doi** [comment]
You may have missed this message: https://github.com/jackun/USBqemu-wheel/issues/30#issuecomment-641177759

**Comment by Florin9doi** [comment]
#4415 is merged. Anything left to do?
